src/mentionui.js

```diff
@@ -61,6 +70,55 @@ export default class MentionUI extends Plugin {
 		this._mentionsConfigurations = new Map();
 
 		editor.config.define( 'mention', { feeds: [] } );
+
+		this._requestFeedDebounced = debounce( ( marker, feedText ) => {
+			this._getFeed( marker, feedText )
+				.catch( error => {
+					if ( error.feedDiscarded ) {
+						// Do nothing on discarded feeds as they might come after another one was requested.
+						return;
+					}
+
+					// At this point something bad happened - most likely unreachable host or some feed callback error.
+					// So cleanup marker, remove the UI and...
+					this._hideUIAndRemoveMarker();
+
+					// ...log warning.
+					/**
+					 * The callback used for obtaining mention autocomplete feed thrown and error and the mention UI was hidden or
+					 * not displayed at all.
+					 *
+					 * @error mention-feed-callback-error
+					 */
+					log.warn( 'mention-feed-callback-error: Could not obtain mention autocomplete feed.' );
+				} )
+				.then( feed => {
+					// Do nothing if :
+					// - feed was discarded or empty feed was passed.
+					// - if the marker is not in the document - the selection might have already changed.
+					if ( !feed || !this.editor.model.markers.has( 'mention' ) ) {
+						return;
+					}
+
+					// Remove old entries.
+					this._items.clear();
+
+					for ( const feedItem of feed ) {
+						const item = typeof feedItem != 'object' ? { id: feedItem, text: feedItem } : feedItem;
+
+						this._items.add( { item, marker } );
+					}
+
+					const markerMarker = editor.model.markers.get( 'mention' );
+
+					if ( this._items.length && markerMarker ) {
+						this._showUI( markerMarker );
```

Comment:
What if the UI was already visible before this?

Reply:
I've renamed the method to make it clear what it does.